openadapt/ml/huggingface_models/generic_provider.py

```diff
@@ -0,0 +1,49 @@
+"""Generic CompletionProvider module."""
+
+from transformers import pipeline
+
+from openadapt.ml.base_provider import (
+    Availability,
+    Capability,
+    CompletionProvider,
+    Modality,
+)
+
+
+class GenericHuggingFaceProvider(CompletionProvider):
+    """GenericHuggingFaceProvider class."""
+
+    Name: str = "Generic HuggingFace Provider"
+    Capabilities: list[Capability] = [
+        Capability.TRAINING,
+        Capability.TUNING,
+        Capability.INFERENCE,
+    ]
+    Modalities: list[Modality] = [Modality.TEXT]
+    Availabilities: list[Availability] = [Availability.HOSTED]
+
+    def infer(
+        self,
+        prompt: str,
+        model_path: str,
+        task_description: str,
```

Comment:
Please rename to `huggingface_task` to differentiate between https://github.com/OpenAdaptAI/OpenAdapt/blob/main/openadapt/models.py#L46